core/tests/test_utils.py

```diff
@@ -3596,6 +3602,44 @@ def publish_collection(self, owner_id: str, collection_id: str) -> None:
         committer = user_services.get_user_actions_info(owner_id)
         rights_manager.publish_collection(committer, collection_id)
 
+    def add_explorations_to_story(
+        self, topic_id: str, story_id: str, exp_ids: List[str]
+    ) -> None:
+        """Appends a story node for each exploration id given in exp_ids to the
+        story with story_id. Each story node is assigned its exp id and then it
+        is published.
+
+        Args:
+            topic_id: str. ID of the topic that contains the story.
+            story_id: str. ID of the story containing the new node.
+            exp_ids: list(str). IDs of the exploration.
+        """
+        story = story_fetchers.get_story_by_id(story_id)
+        change_list = []
+
+        node_id = story.story_contents.next_node_id
+        for exp_id in exp_ids:
+            change_list.extend([
+                story_domain.StoryChange({
+                    'cmd': story_domain.CMD_ADD_STORY_NODE,
+                    'node_id': node_id,
+                    'title': node_id
+                }),
+                story_domain.StoryChange({
+                    'cmd': story_domain.CMD_UPDATE_STORY_NODE_PROPERTY,
+                    'property_name':
+                        story_domain.STORY_NODE_PROPERTY_EXPLORATION_ID,
+                    'node_id': node_id,
+                    'old_value': None,
+                    'new_value': exp_id
+                })])
+            node_id = story_domain.StoryNode.get_incremented_node_id(node_id)
+
+        if len(change_list) > 0:
+            topic_services.update_story_and_topic_summary(
+                feconf.SYSTEM_COMMITTER_ID, story_id, change_list,
+                'Linked explorations to story ' + story_id, topic_id)
```

Comment:
Generally we prefer string interpolation: 'Linked explorations to story %s' % story_id

Reply:
Done.